lib/connection/statement.js

```diff
@@ -1146,6 +1146,13 @@ function createFnStreamRows(statement, context)
           ErrorCodes.ERR_STMT_STREAM_ROWS_INVALID_FETCH_AS_STRING_VALUES,
           JSON.stringify(fetchAsString[invalidValueIndex]));
       }
+      // check for invalid rowMode
+      var rowMode = options.rowMode;
+      if (Util.exists(rowMode))
+      {
+        Errors.checkArgumentValid(['array', 'object'].includes(rowMode),
+          ErrorCodes.ERR_STMT_STREAM_ROWS_INVALID_ROW_MODE);
```

Comment:
1. should we have separate consts when error happens on connection or statement configuration level?
2. can we extract common method for such validation?

Reply:
it is still happenning - can we create a method in RowMode to check this value and throw error is needed?